# 1. Import libraries and datasets

## 1.1 Import libraries

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sb
# import warnings
# warnings.filterwarnings('ignore')

## 1.2 Import datasets

In [ ]:
flipkart_clothes = pd.read_csv('/content/flipkart_reviews_clothes.csv')
flipkart_electronics = pd.read_csv('/content/flipkart_reviews_electronics.csv')
flipkart_mobiles = pd.read_csv('/content/flipkart_reviews_mobiles.csv')

## 1.3 flipkart clothes dataset - test dataset

### 1.3.1 Check head
- To have overall view of the dataset.
- To check for data types just by looking at the dataset.
- Check if use of regular experession is required or not.

In [ ]:
flipkart_clothes.head()

reviews  \
0                                        5Satisfying   
1                                              3Good   
2                      2This product quality is poor   
3                                         3Very nice   
4  5Under this price rage this product is very ni...   

                                          sentiments  ratings  
0                                         Satisfying        5  
1                                               Good        3  
2                       This product quality is poor        2  
3                                          Very nice        3  
4  Under this price rage this product is very nic...        5

### 1.3.2 Check shape of dataset

In [ ]:
flipkart_clothes.shape

(5644, 3)

### 1.3.3 Check for data type and memory

In [ ]:
flipkart_clothes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5644 entries, 0 to 5643
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   reviews     5644 non-null   object
 1   sentiments  5644 non-null   object
 2   ratings     5644 non-null   int64 
dtypes: int64(1), object(2)
memory usage: 132.4+ KB


### 1.3.4 Check for null values

In [ ]:
flipkart_clothes.isnull().sum()

reviews       0
sentiments    0
ratings       0
dtype: int64

### 1.3.5 Check for duplicates

In [ ]:
flipkart_clothes.duplicated().sum()

0

### 1.3.6 Check for integrity of variable
- Use describe function.

In [ ]:
flipkart_clothes["ratings"].describe()

count    5644.000000
mean        3.947378
std         1.391818
min         1.000000
25%         3.000000
50%         5.000000
75%         5.000000
max         5.000000
Name: ratings, dtype: float64

### 1.3.7 Check for imbalance
- Use value counts.

In [ ]:
pd.concat([flipkart_clothes["ratings"].value_counts(), flipkart_clothes["ratings"].value_counts(normalize=True)], axis=1)


count  proportion
ratings                   
5         2991    0.529943
4         1034    0.183203
1          665    0.117824
3          615    0.108965
2          339    0.060064

## 1.4 flipkart mobiles dataset - train dataset

### 1.4.1 Check head
- To have overall view of the dataset.
- To check for data types just by looking at the dataset.
- Check if use of regular experession is required or not.

In [ ]:
flipkart_mobiles.head()

reviews           sentiments  \
0                            everything is very good            Just wow!   
1                             Camera good\nDisplay 👍  Best in the market!   
2  Camera is very good and battery backup is very...       Simply awesome   
3                                    Awesome product          Good choice   
4                                            Awesome      Worth the money   

   ratings  
0      5.0  
1      5.0  
2      5.0  
3      4.0  
4      4.0

### 1.4.2 Check shape of dataset

In [ ]:
flipkart_mobiles.shape

(15850, 3)

### 1.4.3 Check for data type and memory

In [ ]:
flipkart_mobiles.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15850 entries, 0 to 15849
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   reviews     15850 non-null  object 
 1   sentiments  15850 non-null  object 
 2   ratings     15850 non-null  float64
dtypes: float64(1), object(2)
memory usage: 371.6+ KB


### 1.4.4 Check for null values

In [ ]:
flipkart_mobiles.isnull().sum()

reviews       0
sentiments    0
ratings       0
dtype: int64

### 1.4.5 Check for duplicates

In [ ]:
flipkart_mobiles.duplicated().sum()

0

### 1.4.6 Check for integrity of variable
- Use describe function.

In [ ]:
flipkart_mobiles["ratings"].describe()

count    15850.000000
mean         4.415142
std          1.014948
min          1.000000
25%          4.000000
50%          5.000000
75%          5.000000
max          5.000000
Name: ratings, dtype: float64

### 1.4.7 Check for imbalance
- Use value counts.

In [ ]:
pd.concat([flipkart_mobiles["ratings"].value_counts(), flipkart_mobiles["ratings"].value_counts(normalize=True)], axis=1)


count  proportion
ratings                   
5.0      10424    0.657666
4.0       3288    0.207445
3.0       1121    0.070726
1.0        689    0.043470
2.0        328    0.020694

## 1.5 flipkart electronics dataset - validation dataset

### 1.5.1 Check head
- To have overall view of the dataset.
- To check for data types just by looking at the dataset.
- Check if use of regular experession is required or not.

In [ ]:
flipkart_electronics.head()

reviews           sentiments  \
0  If anybody is planning to add a soundbar to th...   Highly recommended   
1  Sound quality is superb\nBass also is good\nBe...           Delightful   
2  I have been researching about soundbar\nSound ...         Nice product   
3  A perfect soundbar in this segment. Superb sou...  Best in the market!   
4                         Value for money..Must Buy👏            Just wow!   

   ratings  
0        5  
1        4  
2        4  
3        5  
4        5

### 1.5.2 Check shape of dataset

In [ ]:
flipkart_electronics.shape

(15684, 3)

### 1.5.3 Check for data type and memory

In [ ]:
flipkart_electronics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15684 entries, 0 to 15683
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   reviews     15684 non-null  object
 1   sentiments  15684 non-null  object
 2   ratings     15684 non-null  int64 
dtypes: int64(1), object(2)
memory usage: 367.7+ KB


### 1.5.4 Check for null values

In [ ]:
flipkart_electronics.isnull().sum()

reviews       0
sentiments    0
ratings       0
dtype: int64

### 1.5.5 Check for duplicates

In [ ]:
flipkart_electronics.duplicated().sum()

0

### 1.5.6 Check for integrity of variable
- Use describe function.

In [ ]:
flipkart_electronics["ratings"].describe()

count    15684.000000
mean         4.277480
std          1.147992
min          1.000000
25%          4.000000
50%          5.000000
75%          5.000000
max          5.000000
Name: ratings, dtype: float64

### 1.5.7 Check for imbalance
- Use value counts.

In [ ]:
pd.concat([flipkart_electronics["ratings"].value_counts(), flipkart_electronics["ratings"].value_counts(normalize=True)], axis=1)


count  proportion
ratings                   
5         9580    0.610814
4         3375    0.215187
3         1265    0.080655
1         1035    0.065991
2          429    0.027353

# 2. Combine flipkart_electronics and flipkart_mobiles dataset
- 15,850 rows from flipkart_electronics dataset.
- 10,000 rows from flipkart_mobiles dataset.

In [ ]:
print("Shape of flipkart_electronics dataset:", flipkart_electronics.shape)
print("Shape of flipkart_mobiles dataset:    ", flipkart_mobiles.shape)

Shape of flipkart_electronics dataset: (15684, 3)
Shape of flipkart_mobiles dataset:     (15850, 3)


## 2.1 Sample 10,000 data points from flipkart_electronics

In [ ]:
sample_elec = flipkart_electronics.sample(n=10000, replace=False, random_state=6)

In [ ]:
sample_elec.shape

(10000, 3)

## 2.2 Add the sampled dataset to flipkart_mobiles dataset

In [ ]:
flipkart_elec_mob = pd.concat([sample_elec, flipkart_mobiles], axis=0)

In [ ]:
flipkart_elec_mob.head()

reviews  \
8244                   All is well\nBest and comfortable   
14302           Awesome ear buds\nAll over fine and good   
7999   Battery charging issue. Need to press earphone...   
3442   I really like the design and lighting effects....   
9596                     Sound and bass quality is good.   

                      sentiments  ratings  
8244         Best in the market!      5.0  
14302          Worth every penny      5.0  
7999   Expected a better product      2.0  
3442       Mind-blowing purchase      5.0  
9596        Good quality product      4.0

In [ ]:
flipkart_elec_mob.shape

(25850, 3)

In [ ]:
flipkart_elec_mob.info()

<class 'pandas.core.frame.DataFrame'>
Index: 25850 entries, 8244 to 15849
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   reviews     25850 non-null  object 
 1   sentiments  25850 non-null  object 
 2   ratings     25850 non-null  float64
dtypes: float64(1), object(2)
memory usage: 807.8+ KB


In [ ]:
flipkart_elec_mob.duplicated().sum()

330

***Observations:***
- We will keep duplicates for now.
- Because as it is only unique words will be indexed, so even if there are duplicate they will be counted only once.

## 2.3 Filtering out sampled datapoints from flipkart_electronics dataset

In [ ]:
flipkart_electronics = (
                          flipkart_electronics
                          .loc[
                              ~flipkart_electronics
                              .index
                              .isin(sample_elec.index)
                              ]
                       )

In [ ]:
flipkart_electronics.shape

(5684, 3)

In [ ]:
print("Shape of flipkart_validation_reviews dataset:", flipkart_electronics.shape)
print("Shape of flipkart_train_reviews dataset:     ", flipkart_elec_mob.shape)

Shape of flipkart_validation_reviews dataset: (5684, 3)
Shape of flipkart_train_reviews dataset:      (25850, 3)


## 2.4 Check info of combined dataset

In [ ]:
print("Data information of flipkart_mobiles dataset:")
print()
print(flipkart_elec_mob.info())
print()
print()
print("Data information of flipkart_electronics dataset:")
print()
print(flipkart_electronics.info())



Data information of flipkart_mobiles dataset:

<class 'pandas.core.frame.DataFrame'>
Index: 25850 entries, 8244 to 15849
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   reviews     25850 non-null  object 
 1   sentiments  25850 non-null  object 
 2   ratings     25850 non-null  float64
dtypes: float64(1), object(2)
memory usage: 807.8+ KB
None


Data information of flipkart_electronics dataset:

<class 'pandas.core.frame.DataFrame'>
Index: 5684 entries, 0 to 15681
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   reviews     5684 non-null   object
 1   sentiments  5684 non-null   object
 2   ratings     5684 non-null   int64 
dtypes: int64(1), object(2)
memory usage: 177.6+ KB
None


***Observations:***
- There are 3 columns in the datasets,
  - reviews
  - sentiments
  - ratings
- There are no null values in the datasets.